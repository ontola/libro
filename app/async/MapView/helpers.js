import * as fa from 'fontawesome';
import Circle from 'ol/geom/Circle';
import ImageState from 'ol/ImageState';
import { toContext } from 'ol/render';
import CircleStyle from 'ol/style/Circle';
import Fill from 'ol/style/Fill';
import Icon from 'ol/style/Icon';
import { get as getIconImage } from 'ol/style/IconImage';
import Stroke from 'ol/style/Stroke';
import Style from 'ol/style/Style';
import Text from 'ol/style/Text';
import { useLayoutEffect, useMemo } from 'react';

import {
  isFontAwesomeIRI,
  normalizeFontAwesomeIRI,
} from '../../helpers/iris';

const ANCHOR_Y_BOTTOM = 1;
const ANCHOR_X_CENTER = 0.5;
const CIRCLE_RADIUS = 12;
const CIRCLE_SIZE = 13;
const ICON_X = CIRCLE_SIZE;
const ICON_Y = CIRCLE_SIZE;
const IMG_SIZE = 26;

const iconCache = {};

const getImage = (image) => {
  const img = getIconImage(null, image, IMG_SIZE, true, ImageState.IDLE, null);
  img.load();

  return img;
};

const generateImageStyle = (image, highlight, count, theme) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  if (highlight) {
    canvasCtx.globalAlpha = 0.6;
  }
  const img = getImage(image);
  canvasCtx.drawImage(img.getImage(), 0, 0, IMG_SIZE, IMG_SIZE);
  canvasCtx.globalAlpha = 1;

  const numberText = count > 1 && (
    new Text({
      fill: new Fill({ color: 'white' }),
      font: `bold 16px ${theme.typography.h1.fontFamily}`,
      offsetX: CIRCLE_RADIUS,
      text: count.toString(),
    })
  );

  return {
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
    text: numberText,
  };
};

const generateFontAwesomeStyle = (image, highlight, count, theme) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  const circleBackground = new Fill({ color: highlight ? '#92a1b5' : '#475668' });
  const circleStroke = new Stroke({
    color: 'white',
    width: 2,
  });
  const circleStyle = new Style({
    fill: circleBackground,
    image: new CircleStyle({
      fill: circleBackground,
      radius: 10,
      stroke: circleStroke,
    }),
    stroke: circleStroke,
  });

  const circle = new Circle([CIRCLE_SIZE, CIRCLE_SIZE], CIRCLE_RADIUS);

  const vectorContext = toContext(canvasCtx, {
    pixelRatio: 1,
    size: [100, 100],
  });
  vectorContext.setStyle(circleStyle);
  vectorContext.drawGeometry(circle);

  let renderText, renderFont;
  if (count > 1) {
    renderText = count;
    renderFont = `bold 16px ${theme.typography.h1.fontFamily}`;
  } else {
    renderText = fa(normalizeFontAwesomeIRI(image));
    renderFont = 'normal 18px FontAwesome';
  }
  /* eslint-disable no-param-reassign */
  canvasCtx.fillStyle = 'white';
  canvasCtx.font = renderFont;
  canvasCtx.textAlign = 'center';
  canvasCtx.textBaseline = 'middle';
  /* eslint-enable no-param-reassign */
  canvasCtx.fillText(renderText, ICON_X, ICON_Y);

  return {
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
  };
};

const generateIconStyle = (image, highlight = false, count, theme) => {
  let style;
  if (isFontAwesomeIRI(image)) {
    style = generateFontAwesomeStyle(image, highlight, count, theme);
  } else {
    style = generateImageStyle(image, highlight, count, theme);
  }

  return new Style(style);
};

const featureCount = (feature) => (
  feature?.get('features')?.length || 0
);

const getStyle = (image, highlight, theme) => (feature) => {
  const count = featureCount(feature);
  const cacheKey = [image, highlight, count].filter(Boolean).join('-');

  if (!iconCache[cacheKey]) {
    const icon = generateIconStyle(image, highlight, count, theme);

    iconCache[cacheKey] = icon;
  }

  return iconCache[cacheKey];
};

export const useImageStyles = (images, theme) => {
  const normalizedImages = useMemo(() => images.map((image) => image?.value || image), []);
  useLayoutEffect(() => {
    normalizedImages.forEach((image) => {
      if (image && !isFontAwesomeIRI(image)) {
        getImage(image);
      }
    });
  }, normalizedImages);

  const hoverStyles = useMemo(() => {
    const result = {};
    normalizedImages.forEach((image) => {
      result[image] = getStyle(image, false, theme);
    });

    return result;
  }, normalizedImages);
  const styles = useMemo(() => {
    const result = {};
    normalizedImages.forEach((image) => {
      result[image] = getStyle(image, true, theme);
    });

    return result;
  }, normalizedImages);

  return {
    hoverStyles,
    styles,
  };
};
