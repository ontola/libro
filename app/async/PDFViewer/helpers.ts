import React from "react";
import { History } from "history";

/** Mapping for consistent component Ids, used by reactivesearch */
export const ids = {
  searchbox: "zoekterm",
  organisaties: "organisaties",
  daterange: "datums",
  type: "type",
  tag: "thema",
  location: "locatie",
};

export function usePersistedState<T>(key: string, initial: T):
  [T, React.Dispatch<React.SetStateAction<T>>] {
  const [value, setValue] = React.useState<T>(() => {
    const storageValue = sessionStorage.getItem(key);
    if (storageValue) {
      return JSON.parse(storageValue) || initial;
    }

    return initial;
  });

  const setPersistedValue = (next: T | React.SetStateAction<T>): void => {
    sessionStorage.setItem(key, JSON.stringify(next));
    setValue(next);
  };
  return [value, setPersistedValue];
}

export function myPersistedState<T>(key: string, initial: T): T {
  const storageValue = sessionStorage.getItem(key);
  if (storageValue === null) {
    return initial;
  }
  return JSON.parse(storageValue);
}

export const getTopicsApiURL = (relUrl: string): URL => {
  const url = new URL(window.location.origin);
  url.pathname = `/topics_api/dev${relUrl}`;

  return url
};

export const getParams = (history: History) => {
  const searchObject = history.location.search;
  const hasParams = searchObject !== "";
  const params = new URLSearchParams(searchObject);

  const currentResourceBase = params.get("showResource");
  let currentResource = null;
  if (currentResourceBase !== null) {
    currentResource = decodeURIComponent(currentResourceBase);
  }

  let currentSearchTerm = params.get(ids.searchbox);
  if (currentSearchTerm) {
    currentSearchTerm = currentSearchTerm.substr(1, currentSearchTerm.length - 2);
  }

  let documentID = null;
  if (currentResource !== null) {
    documentID = currentResource.split("/")[3]
  }

  return {
    currentResource,
    currentSearchTerm,
    hasParams,
    documentID
  };
};
