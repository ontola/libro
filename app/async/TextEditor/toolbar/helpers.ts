import { makeStyles } from '@material-ui/core';
import { isNode, NamedNode, Node } from '@ontologies/core';
import { LinkReduxLRSType, useLRS } from 'link-redux';
import { EventHandler } from 'react';
import { useSlate } from 'slate-react';

export const useMenuItemStyles = makeStyles(({
  button: {
    border: 'none',
    height: 48,
  },
  root: {
    backgroundColor: 'transparent',
  },
}));

export function normalizeAction(
  lrs: LinkReduxLRSType,
  action: Node | EventHandler<any>,
  opts: any,
): EventHandler<any> {
  if (isNode(action)) {
    return (e: Event) => {
      e.preventDefault();

      return lrs.exec(action as NamedNode, opts);
    };
  }

  return action;
}

export const useNormalizeAction = (action: EventHandler<any>, item: any) => {
  const lrs = useLRS();
  const textEditor = useSlate();
  const actionOpts = { item, textEditor };

  return normalizeAction(lrs, action, actionOpts);
};
