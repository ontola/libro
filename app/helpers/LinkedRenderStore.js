/* eslint no-console: 0 */
import LinkedRenderStore, { DataWorkerLoader } from 'link-lib';
import { PropTypes } from 'react';

import DataWorker from 'worker-loader!../workers/DataWorker';
import transformers from './transformers';

import Error from '../components/Error';

const LRS = new LinkedRenderStore();

LRS.onError = Error;

if (typeof window !== 'undefined' && window.Worker) {
  LRS.api.processor = new DataWorkerLoader(DataWorker);
} else {
  transformers.transformers.forEach(
    t => LRS.api.registerTransformer(t.transformer, t.mediaTypes, t.acceptValue)
  );
}

LRS.api.setAcceptForHost('https://argu.dev/', 'application/vnd.api+json');
LRS.api.setAcceptForHost('https://beta.argu.dev/', 'application/vnd.api+json');
LRS.api.setAcceptForHost('https://argu.co/', 'application/vnd.api+json');
LRS.api.setAcceptForHost('https://aod-search.argu.co/', 'application/vnd.api+json');
LRS.api.setAcceptForHost('https://beta.argu.co/', 'application/vnd.api+json');

LRS.namespaces.aod = rdf.Namespace('https://argu.co/ns/od#');
export const NS = LRS.namespaces;

LRS.addOntologySchematics([
  {
    '@id': 'http://www.w3.org/2000/01/rdf-schema#Resource',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/Thing',
    },
  },
  {
    '@id': 'http://www.w3.org/2002/07/owl#Thing',
    'owl:sameAs': {
      '@id': 'http://schema.org/Thing',
    },
  },
  {
    '@id': 'http://wikiba.se/ontology#Entity',
    'owl:sameAs': {
      '@id': 'http://schema.org/Thing',
    },
  },
  {
    '@id': 'http://schema.org/CreativeWork',
    '@type': 'rdfs:Class',
    'http://purl.org/dc/terms/source': {
      '@id': 'http://www.w3.org/wiki/WebSchemas/SchemaDotOrgSources#source_rNews',
    },
    'rdfs:comment': 'The most generic kind of creative work, including books, movies, photographs, software programs, etc.',
    'rdfs:label': 'CreativeWork',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/Thing',
    },
  },
  {
    '@id': 'http://schema.org/Argument',
    '@type': 'rdfs:Class',
    'http://purl.org/dc/terms/source': {
      '@id': 'http://www.w3.org/wiki/WebSchemas/SchemaDotOrgSources#source_rNews',
    },
    'rdfs:comment': 'The most generic kind of creative work, including books, movies, photographs, software programs, etc.',
    'rdfs:label': 'CreativeWork',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/Comment',
    },
  },
  {
    '@id': 'http://schema.org/Comment',
    '@type': 'rdfs:Class',
    'rdfs:comment': "A comment on an item - for example, a comment on a blog post. The comment's content is " +
    'expressed via the <a class="localLink" href="/text">text</a> property, and its topic ' +
    'via <a class="localLink" href="/about">about</a>, properties shared with all CreativeWorks.',
    'rdfs:label': 'Comment',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/CreativeWork',
    },
  },
  {
    '@id': 'https://argu.co/ns/core#Question',
    '@type': 'rdfs:Class',
    'rdfs:comment': '',
    'rdfs:label': 'Challenge',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/CreativeWork',
    },
  },
  {
    '@id': 'http://schema.org/Thing',
    '@type': 'rdfs:Class',
    'rdfs:comment': 'The most generic type of item.',
    'rdfs:label': 'Thing',
  },
  {
    '@id': 'http://schema.org/Person',
    '@type': 'rdfs:Class',
    'http://purl.org/dc/terms/source': {
      '@id': 'http://www.w3.org/wiki/WebSchemas/SchemaDotOrgSources#source_rNews',
    },
    'http://www.w3.org/2002/07/owl#equivalentClass': {
      '@id': 'http://xmlns.com/foaf/0.1/Person',
    },
    'rdfs:comment': 'A person (alive, dead, undead, or fictional).',
    'rdfs:label': 'Person',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/Thing',
    },
  },
  {
    '@id': 'http://xmlns.com/foaf/0.1/name',
    'owl:sameAs': {
      '@id': 'http://schema.org/name',
    },
  },
  {
    '@id': 'https://argu.co/ns/od#title',
    'owl:sameAs': {
      '@id': 'http://schema.org/name',
    },
  },
  {
    '@id': 'https://argu.co/ns/od#Motie',
    'owl:sameAs': {
      '@id': 'https://argu.co/ns/core#Motion',
    },
  },
  {
    '@id': 'https://argu.co/ns/od#Amendement',
    'owl:sameAs': {
      '@id': 'https://argu.co/ns/core#Motion',
    },
  },
  {
    '@id': 'https://argu.co/ns/od#createdAt',
    'owl:sameAs': {
      '@id': 'http://schema.org/dateCreated',
    },
  },
  {
    '@id': 'https://argu.co/ns/core#Motion',
    '@type': 'rdfs:Class',
    'rdfs:label': 'Motion',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/CreativeWork',
    },
  },
  {
    '@id': 'https://argu.co/ns/core#Argument',
    '@type': 'rdfs:Class',
    'rdfs:label': 'Argument',
    'rdfs:subClassOf': {
      '@id': 'http://schema.org/CreativeWork',
    },
  },
  {
    '@id': 'https://argu.co/ns/od#text',
    'owl:sameAs': {
      '@id': 'http://schema.org/text',
    },
  },
  {
    '@id': 'https://argu.co/ns/core#VoteCollection',
    'owl:sameAs': {
      '@id': 'https://argu.co/ns/core#Collection',
    },
  },
]);

const linkedPropVal = PropTypes.oneOfType([
  PropTypes.object,
  PropTypes.string,
  undefined,
]);

export {
  linkedPropVal,
};

export default LRS;
window.LRS = LRS;
