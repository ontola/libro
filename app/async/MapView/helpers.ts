import { Theme } from '@material-ui/core';
import fa from 'fontawesome';
import { Feature } from 'ol';
import Circle from 'ol/geom/Circle';
import { toContext } from 'ol/render';
import CircleStyle from 'ol/style/Circle';
import Fill from 'ol/style/Fill';
import Icon from 'ol/style/Icon';
import Stroke from 'ol/style/Stroke';
import Style from 'ol/style/Style';

import { isFontAwesomeIRI, normalizeFontAwesomeIRI } from '../../helpers/iris';

const ANCHOR_Y_BOTTOM = 1;
const ANCHOR_X_CENTER = 0.5;
const CIRCLE_RADIUS = 12;
const CIRCLE_SIZE = 13;
const ICON_X = CIRCLE_SIZE;
const ICON_Y = CIRCLE_SIZE;
const IMG_SIZE = 26;

const iconCache: { [k: string]: Style } = {};

const drawFontAwesomeIcon = (
  canvasCtx: CanvasRenderingContext2D,
  text: string,
  highlight: boolean,
  count: number,
  theme: any,
) => {
  const circleBackground = new Fill({ color: highlight ? '#92a1b5' : '#475668' });
  const circleStroke = new Stroke({
    color: 'white',
    width: 2,
  });
  const circleStyle = new Style({
    fill: circleBackground,
    image: new CircleStyle({
      fill: circleBackground,
      radius: 10,
      stroke: circleStroke,
    }),
    stroke: circleStroke,
  });

  const circle = new Circle([CIRCLE_SIZE, CIRCLE_SIZE], CIRCLE_RADIUS);

  const vectorContext = toContext(canvasCtx, {
    pixelRatio: 1,
    size: [100, 100],
  });
  vectorContext.setStyle(circleStyle);
  vectorContext.drawGeometry(circle);

  let renderText;
  let renderFont;
  if (count > 1) {
    renderText = count.toString();
    renderFont = `bold 16px ${theme.typography.h1.fontFamily}`;
  } else {
    renderText = text;
    renderFont = 'normal 18px FontAwesome';
  }
  /* eslint-disable no-param-reassign */
  canvasCtx.fillStyle = 'white';
  canvasCtx.font = renderFont;
  canvasCtx.textAlign = 'center';
  canvasCtx.textBaseline = 'middle';
  /* eslint-enable no-param-reassign */
  canvasCtx.fillText(renderText, ICON_X, ICON_Y);
};

const generateIconStyle = (image: string, highlight = false, count: number, theme: any) => {
  const canvas = document.createElement('canvas');
  const canvasCtx = canvas.getContext('2d');

  if (canvasCtx && isFontAwesomeIRI(image)) {
    drawFontAwesomeIcon(canvasCtx, fa(normalizeFontAwesomeIRI(image)), highlight, count, theme);
  }

  return new Style({
    image: new Icon({
      anchor: [ANCHOR_X_CENTER, ANCHOR_Y_BOTTOM],
      img: canvas,
      imgSize: [IMG_SIZE, IMG_SIZE],
    }),
  });
};

const featureCount = (feature: Feature) => (
  feature?.get('features')?.length || 0
);

const getStyle = (image: string, highlight: boolean, theme: any) => (feature: Feature) => {
  const count = featureCount(feature);
  const cacheKey = [image, highlight, count].filter(Boolean).join('-');

  if (!iconCache[cacheKey]) {
    const icon = generateIconStyle(image, highlight, count, theme);

    iconCache[cacheKey] = icon;
  }

  return iconCache[cacheKey];
};

export const getStyles = (image: string, theme: Theme): {
  hoverStyle: (feature: Feature) => Style,
  style: (feature: Feature) => Style,
} => {
  const style = getStyle(image, false, theme);
  const hoverStyle = getStyle(image, true, theme);

  return {
    hoverStyle,
    style,
  };
};
