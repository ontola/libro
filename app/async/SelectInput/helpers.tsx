import { FilterOptionsState } from '@material-ui/lab';
import { SomeTerm, isLiteral } from '@ontologies/core';
import * as rdfx from '@ontologies/rdf';
import * as schema from '@ontologies/schema';
import { SomeNode } from 'link-lib';
import { Resource, useLRS } from 'link-redux';
import React from 'react';
import { defineMessages } from 'react-intl';

import { entityIsLoaded } from '../../helpers/data';
import { isResource } from '../../helpers/types';
import ontola from '../../ontology/ontola';

const messages = defineMessages({
  noMatchingItems: {
    defaultMessage: 'No matching items',
    id: 'https://app.argu.co/i18n/forms/select/noMatchingItems',
  },
  typeToSearch: {
    defaultMessage: 'Type to start searching',
    id: 'https://app.argu.co/i18n/forms/select/typeToSearch',
  },
});

const stripDiacritics = (str: string) => (
  typeof str.normalize !== 'undefined'
    ? str.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
    : str
);

export const filterOptions = (
  options: SomeTerm[],
  { inputValue, getOptionLabel }: FilterOptionsState<SomeTerm>,
): SomeTerm[] => {
  const input = stripDiacritics(inputValue.toLowerCase());

  return options.filter((option) => {
    const candidate = stripDiacritics(getOptionLabel(option).toLowerCase());

    return candidate.indexOf(input) > -1;
  });
};

export const useItemToString = (): ((i: SomeTerm | undefined | null) => string) => {
  const lrs = useLRS();

  return React.useCallback((item: SomeTerm | undefined | null) => {
    if (typeof item === 'undefined' || item === null) {
      return '';
    }

    if (!isResource(item)) {
      return item.value;
    }

    if (!entityIsLoaded(lrs, item)) {
      return 'Loading';
    }

    const itemClass = lrs.getResourceProperty<SomeNode>(item, rdfx.type);
    const itemClassDisplayProp = lrs.getResourceProperty<SomeNode>(
      itemClass,
      ontola['forms/inputs/select/displayProp'],
    );
    const classDisplayProp = itemClassDisplayProp ?? schema.name;
    const label = lrs.getResourceProperty(item, [classDisplayProp, schema.name]);

    return label?.value ?? item.value;
  }, [lrs]);
};

export function emptyMessage(
  fmt: (args: any) => string,
  searchable: boolean,
  currentValue: string,
): string {
  if (searchable) {
    if (currentValue && currentValue.length > 0) {
      return fmt(messages.noMatchingItems);
    }

    return fmt(messages.typeToSearch);
  }

  return fmt(messages.noMatchingItems);
}

export const renderOption = (item: SomeTerm): JSX.Element => {
  if (isLiteral(item.termType)) {
    return (
      <option
        className="Field__list-element"
        key={item.value}
        value={item.value}
      >
        {item.value}
      </option>
    );
  }

  return (
    <Resource
      element="div"
      key={item.value}
      subject={item}
    />
  );
};
