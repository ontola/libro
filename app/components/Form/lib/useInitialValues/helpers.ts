import {
  NamedNode,
  SomeTerm,
  isNode,
} from '@ontologies/core';
import * as sh from '@ontologies/shacl';
import { SomeNode } from 'link-lib/dist-types/types';
import { LinkReduxLRSType } from 'link-redux';

import { conditionalFormFieldsPath, formFieldsPath } from '../../../../helpers/diggers';
import { inputValueFromStorage } from '../../../../helpers/persistence';
import form from '../../../../ontology/form';

export const getFormFields = (lrs: LinkReduxLRSType, actionBody: SomeNode): SomeNode[] =>
  (lrs.dig(actionBody, formFieldsPath).concat(lrs.dig(actionBody, conditionalFormFieldsPath))).filter(isNode);

export const rawFormObjectValue = (lrs: LinkReduxLRSType, field: SomeNode, object: SomeNode, sessionStore: Storage | undefined, formID: string, nested: boolean): SomeTerm[] => {
  const path = lrs.getResourceProperty<NamedNode>(field, sh.path);

  if (!path) {
    return [];
  }

  const defaultValue = lrs.getResourceProperties(field, form.defaultValue);
  const valueFromStorage = inputValueFromStorage(sessionStore, path, object, formID, nested);

  if (valueFromStorage) {
    return valueFromStorage;
  }

  return defaultValue.length > 0 ? defaultValue : lrs.getResourceProperties(object, path);
};
