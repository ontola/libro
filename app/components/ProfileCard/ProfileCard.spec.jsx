import React from 'react';
import rdf from 'rdflib';

import { NS } from '../../helpers/LinkedRenderStore';

import ProfileCardConnected, { ProfileCard, propTypes } from './index';

const resource = new rdf.NamedNode('http://example.com/profile/1');
const resources = {

  [resource]: {
    [NS.schema('name')]: new rdf.Literal('Title'),
    [NS.schema('description')]: new rdf.Literal('bio'),
  },
};

describeView('ProfileCard', [], resources, resource, () => {
  set('ch', () => <ProfileCardConnected />);

  describe('with link', () => {
    it('should render', () => {
      expect(subject.find(ProfileCardConnected)).toMatchSnapshot();
      expect(subject.find(marker()));
    });

    it('should display the title', () => {
      expect(subject.find(marker('head', 'name'))).toHaveText('Title');
    });

    it('should display the bio', () => {
      expect(subject.find(marker('bio'))).toHaveText('bio');
    });

    it('should display the footer', () => {
      expect(subject.find(marker('footer'))).toBePresent();
    });
  });

  describe('without link', () => {
    set('ch', () => (
      <ProfileCard
        bio="classic bio"
        name="classic name"
        party="classic party"
      />
    ));


    it('should display the title', () => {
      expect(subject.find(marker('head', 'name'))).toHaveText('classic name');
    });

    it('should display the bio', () => {
      expect(subject.find(marker('bio'))).toHaveText('classic bio');
    });

    it('should display the footer', () => {
      expect(subject.find(marker('footer'))).toBePresent();
    });

    it('should display the party', () => {
      expect(subject.find(marker('head', 'party'))).toHaveText('classic party');
    });
  });
}, { propTypes });
