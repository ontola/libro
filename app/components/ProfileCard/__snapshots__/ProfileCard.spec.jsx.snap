// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard with link should render 1`] = `
<linkedSubject>
  <ProfileCard
    full={false}
    loading={false}
    subject={
      NamedNode {
        "termType": "NamedNode",
        "value": "http://example.com/profile/1",
      }
    }
  >
    <section
      className="ProfileCard ProfileCard--small"
      data-test="ProfileCard"
    >
      <div
        className="ProfileCard__head"
        data-test="ProfileCard-head"
      >
        <div
          className="ProfileCard__title"
        >
          <Heading
            data-test="ProfileCard-head-name"
            size="2"
          >
            <h2
              className="Heading null"
              role="heading"
            >
              Title
            </h2>
          </Heading>
        </div>
      </div>
      <p
        className="ProfileCard__bio"
        data-test="ProfileCard-bio"
      >
        bio
      </p>
      <div
        className="ProfileCard__foot"
        data-test="ProfileCard-footer"
      >
        <div
          className="ProfileCard__stats"
        >
          <div
            className="ProfileCard__stat"
          >
            Ideeen
            <span
              className="ProfileCard__stat-value"
            >
              124
            </span>
          </div>
          <div
            className="ProfileCard__stat"
          >
            Opinies
            <span
              className="ProfileCard__stat-value"
            >
              543
            </span>
          </div>
        </div>
        <div
          className="ProfileCard__buttons"
        >
          <Button
            active={false}
            icon="tachometer"
            narrow={false}
            onClick={[Function]}
            plain={false}
            small={true}
            theme="default"
            type="button"
          >
            <button
              className="Button Button--has-icon Button--small Button--default"
              data-test="Button-button"
              onClick={[Function]}
              type="button"
            >
              <FontAwesome
                className="Button__icon"
                data-test="Button-icon"
                name="tachometer"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-tachometer Button__icon"
                  data-test="Button-icon"
                />
              </FontAwesome>
              <span
                className="Button__label"
              >
                Vergelijk
              </span>
            </button>
          </Button>
        </div>
      </div>
      <div
        className="ProfileCard__buttons"
      >
        <Button
          active={false}
          icon="tachometer"
          narrow={false}
          onClick={[Function]}
          plain={false}
          small={true}
          theme="default"
          type="button"
        >
          <button
            className="Button Button--has-icon Button--small Button--default"
            data-test="Button-button"
            onClick={[Function]}
            type="button"
          >
            <FontAwesome
              className="Button__icon"
              data-test="Button-icon"
              name="tachometer"
            >
              <span
                aria-hidden={true}
                className="fa fa-tachometer Button__icon"
                data-test="Button-icon"
              />
            </FontAwesome>
            <span
              className="Button__label"
            >
              Vergelijk
            </span>
          </button>
        </Button>
      </div>
    </section>
  </ProfileCard>
</linkedSubject>
`;
